EIPS/eip-5719.md

```diff
@@ -0,0 +1,86 @@
+---
+eip: 5719
+title: Signature replacement interface
+description: Non-interactive replacing of smart contract wallet signatures that became stale due to configuration changes.
+author: Agustin Aguilar (@Agusx1211)
+discussions-to: https://ethereum-magicians.org/t/erc-signature-replacing-for-smart-contract-wallets/11059
+status: Draft
+type: Standards Track
+category: ERC
+created: 2022-09-26
+requires: 1271
+---
+
+## Abstract
+
+A proposal for a standard method to re-encode or replace smart contract wallet signatures at any point after the signature was produced. Wallets implementing this EIP can expose a URI that clients can use to replace an invalid signature with a valid one.
+
+## Motivation
+
+In contrast to EOA signatures, smart contract wallet signatures can become invalid at any point in time; this poses a challenge to protocols that rely on signatures remaining valid for extended periods of time.
+
+A signature may need to be mutated due to one of the following scenarios:
+
+1) The wallet removes a signer that contributed to signing the initial message.
+2) The wallet uses a Merkle tree to store signers, adding a new signer.
+3) The wallet uses a Merkle tree to store signatures, adding new signatures.
+4) The wallet is updated to a new implementation, and the signature schema changes.
+
+Signature replacement should be possible in a non-interactive way, since the wallet that originally signed the message may not be available when the signature needs to be validated (E.g. the settlement of a trade in an exchange that uses an off-chain orderbook).
+
+## Specification
```

Comment:
Do you not want to use RFC 8174?

Reply:
I guess it isn't technically required to include this notice, so I'll still approve this PR, but I would ask the authors to include it since the body of the EIP seems to use `MUST`.

```suggestion
## Specification

The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
```